### DISCUSSION TITLE
json_decode

### DISCUSSION THREAD

**Discussion by jcgmalave** [filamentphp/filament#2053] [Help]
can i apply json:decode to a field to display its contents in TableBuilder

**Comment by danharrin** [comment]
```php
->formatStateUsing(fn ($state) => json_decode($state))
```

**Reply by jcgmalave** [reply]
THANKS!!!! no i get this error: Array to string conversion in protected function mutateArrayState(array $state): string

**Reply by danharrin** [reply]
Hm, maybe try instead `->getStateUsing(fn ($record) => json_decode($record->FIELD_NAME_HERE))`

**Reply by jcgmalave** [reply]
now i get this: json_decode(): Argument #1 ($json) must be of type string, array given

**Reply by ryangjchandler** [reply]
Have you got a `json` or `array` cast on the field? If so, it'll already be an array?

**Reply by danharrin** [reply]
You're trying to render the JSON as text, right? Sounds like you need `json_encode()` instead

**Reply by jcgmalave** [reply]
THANKS GUYS, I LOVE U!!!!
